include/rosparam_shortcuts/parameter.h

```diff
@@ -32,6 +32,7 @@ class Parameter
 	// void setProperties(...);
 
 	// Validity Status
+	const bool isInitialized() const noexcept;  // true when created with default
```

Comment:
Yep, or maybe `isDeclared()`? I guess Parameters could have a lot of different states that we might want to check for. I'm just not sure if this needs to be exposed to the "user", or if it's enough to handle these internally and only provide a public `isValid()` that simply reads as "Can I use this parameter or not?".

Reply:
That is fair.  I think from a use pattern the only case where you would use this is to expose some way for NodeParameters to have a `isValid` interface to ask if all parameters are valid before reading them (so it can know if it'll get an exception).  In that case having only `isValid` makes sense.  The NodeParameters could maybe have an optional timeout on `isValid` where it would retry up until the timeout if any parameters aren't valid (hasn't finished initializing).